lib/ipam/ipam.go

```diff
@@ -854,17 +875,39 @@ func (c ipamClient) ReleaseIPs(ctx context.Context, ips []net.IP) ([]net.IP, err
 		ipsByBlock[cidrStr] = append(ipsByBlock[cidrStr], ip)
 	}
 
-	// Release IPs for each block.
+	// Release IPs for each block. These don't typically compete for resources, so we can do them in parallel
+	// in order to move quickly.
+	type retVal struct {
+		Error       error
+		Unallocated []net.IP
+	}
+	resultChan := make(chan retVal, len(ipsByBlock))
+	log.Infof("Starting %d routines to release IP addresses", len(ipsByBlock))
 	for cidrStr, ips := range ipsByBlock {
 		_, cidr, _ := net.ParseCIDR(cidrStr)
-		unalloc, err := c.releaseIPsFromBlock(ctx, ips, *cidr)
-		if err != nil {
-			log.Errorf("Error releasing IPs: %v", err)
-			return nil, err
+		go func(cidr net.IPNet, ips []net.IP) {
```

Comment:
Should probably set a limit on parallelism here; generally better to have GO_MAX_PROC goroutines working relatively fast than 1000 goroutines all competing for API server resource, I reckon.

I've used this pattern in the past in felix:
```
	sem := semaphore.NewWeighted(int64(maxWorkers))
	for ... {
		if err := sem.Acquire(context.Background(), 1); err != nil {
			// Should only happen if the context finishes.
			log.WithError(err).Panic("Failed to acquire semaphore")
		}
		
		go func() {
			defer sem.Release(1)
			...
		}()
	})
```